### PR TITLE
feat(#3206): Adds a method to sort Quarkus Test Resources

### LINKED ISSUE #3206: Quarkus Test Resources should be sortable
**Description**
Quarkus Test Resources should be sortable so the developer can have a way to define in which order they should be started.

### PR DESCRIPTION
**Pull request by lordofthejars**
Fixes #3206

### CODE REVIEW

**Comment by geoand** [comment]
Can you please squash the change into a single commit @lordofthejars ?

File: test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResourceLifecycleManager.java [hunk 1]
```diff
@@ -32,4 +32,14 @@
      */
     default void inject(Object testInstance) {
     }
+
+    /**
+     * If multiple Test Resources are specified,
+     * this control the order of which they will be executed.
+     * 
+     * @return The order to be executed. The bigger the number, it is executed later.
```
Comment (geoand) [inline test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResourceLifecycleManager.java]: Would probably be better phrased like so: `The larger the number, the later the Resource is invoked`

**Comment by lordofthejars** [comment]
@geoand It can be squashed when merged, there is a Squash and Merge button. Or you said for any other reason?

**Comment by geoand** [comment]
@lordofthejars , when merging PRs Quarkus has elected to always create a merge commit and the Github UI doesn't allow you to both create a merge commit and squash commits to a single one (unless I have missed that).

**Comment by lordofthejars** [comment]
@geoand Maybe you've got configured in this way in the Quarkus repo, but in FMP we are doing this way, merge + squash button.

**Comment by geoand** [comment]
In any case I can't change things, so it's probably best to do what we have been doing for all other PRs.

Thanks

**Comment by gsmet** [comment]
I confirm we can't do Squash + merge commit with the GitHub UI. That's why we ask for the authors to squash things themselves.

Squash + merge is a fast-forward merge. See https://help.github.com/en/articles/about-pull-request-merges#squash-and-merge-your-pull-request-commits for more info.

**Comment by lordofthejars** [comment]
@gsmet @geoand Squashed.

**Review by geoand — APPROVED** [review]

**Pull request merged**